_posts/en/newsletters/2021-03-24-newsletter.md

```diff
@@ -0,0 +1,299 @@
+      taproot script ([tapscript][topic tapscript]).  That hash
+      commitment can be used as part of a [scheme][ruffing scheme] to
+      transition to a new cryptographic algorithm that's safe against
+      QCs, or---if such an algorithm is standardized for Bitcoin before
+      QCs become a threat---it can allow the owner of the bitcoins to
+      immediately transition to the new scheme.  This does only work if
+      individual users create backup tapscript spending paths, if they
+      don't share any public keys (including BIP32 xpubs) involved in
+      those backup paths, and if we learn about a powerful QC before it
+      does too much damage to Bitcoin.
+
+    - *Is the attack realistic?*  One respondent thought a fast QC by the
+      end of the decade [was][fournier progress] "on the wildly
+      optimistic side of projected rate of progress."  Another
+      [noted][towns parallelization] it was a "fairly straightforward
+      engineering challenge" to turn the design for a single slow QC
+      into a farm of QCs that could work in parallel, achieving results
+      in a fraction of the time---making any protection from P2PKH-style
+      hash indirection irrelevant.  A third respondent
+      [proposed][harding bounty] constructing special Bitcoin addresses
+      that could only be spent from by someone making progress on fast
+      QCs; users could voluntarily donate bitcoins to the addresses to
+      create an incentivized early warning system.
+
+    - *We could add a hash-style address after taproot is activated:* if
+      a significant number of users really do think the sudden
+      appearance of a powerful QC is a threat, we could use a follow-up
+      soft fork to [add][dashjr quantum] an alternative P2PKH-style taproot
+      address type that uses hashes.  However, this has consequences
+      that caused several respondents to oppose it:
+
+        - It'll create confusion
+
+        - It'll use more block space
+
+        - It [reduces][poelstra anon] the size of taproot's anonymity
+          set, both directly and when protocols such as [ring signature
+          membership proofs][nick ring] or [Provisions][] are being used
+
+    - *Bandwidth/storage costs versus CPU costs:* it's [possible][rubin
+      recovery] to eliminate the extra 32-byte storage overhead from
+      hash indirection by deriving the public key from a signature and
+      the transaction data it signs, a technique called *key recovery*.
+      Again, this was opposed.  Key recovery requires a [significant amount of CPU][corallo
+      recovery overhead] that would slow down nodes and also prevents
+      the use of schnorr batch validation that can make historic block
+      verification up to three times faster.  It also [makes][poelstra
+      slowdowns] anonymous membership proofs and related techniques both
+      harder to develop and much more CPU intensive.  There
+      [may][poelstra patent story] also be a patent issue.
```

Comment:
"There may also be a patent issue." <- is there a better source for this than Andrew Poelstra's very oblique reference?

Reply:
I've heard about this patent for years, but all I was able to find were claims that it was owned by Certicom (no surprise there, they own over a 100 ECC patents).  I sent Poelstra an email to see if he knows.

Reply:
Added link to the patent Poelstra sent me.  I did a quick skim of it and it looks like it could indeed be related to key recovery, although I'd be surprised if there wasn't prior art before 2007 and Google says the patent is expired (though for non-payment of fees, not because of age; I don't know if non-payment is something they could recover from).  Certicom were super litigious in their prime, though, so I wouldn't blame anyone for being careful about them.

Reply:
Wow, thanks for the extra research! Very interesting.